components/fxa-client/src/push.rs

```diff
@@ -0,0 +1,139 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use crate::{error::*, AccountEvent, FirefoxAccount};
+use serde_derive::Deserialize;
+
+impl FirefoxAccount {
+    /// Handle any incoming push message payload coming from the Firefox Accounts
+    /// servers that has been decrypted and authenticated by the Push crate.
+    ///
+    /// Due to iOS platform restrictions, a push notification must always show UI,
+    /// and therefore we only retrieve 1 command per message.
+    ///
+    /// **💾 This method alters the persisted account state.**
+    pub fn handle_push_message(&mut self, payload: &str) -> Result<Vec<AccountEvent>> {
```

Comment:
For whoever reviews the rust parts of this PR: this is the meat of the patch, the rest of the code is pretty much a bunch of type conversions.